app/controllers/auth/sessions_controller.rb

```diff
@@ -21,6 +21,11 @@ class Auth::SessionsController < Devise::SessionsController
   end
 
   def create
+    user = warden.authenticate(scope: :user)
+    unless user
+      log_failed_authentication(email)
+    end
```

Comment:
This could be shortened as:
```suggestion
  log_failed_authentication(email) unless user_signed_in?
```

But I wonder if this is the right place to do so, or if we should plug into Devise/Warden's error handling.

Also… was it meant to say `email`?